lib/lambda_ethereum_consensus/state_transition/misc.ex

```diff
@@ -58,4 +108,17 @@ defmodule LambdaEthereumConsensus.StateTransition.Misc do
     slots_per_epoch = ChainSpec.get("SLOTS_PER_EPOCH")
     epoch * slots_per_epoch
   end
+
+  @spec bytes_to_uint64(binary()) :: SszTypes.uint64()
+  defp bytes_to_uint64(value) do
+    # Converts a binary value to a 64-bit unsigned integer
+    <<first_8_bytes::binary-size(8), _::binary>> = value
+    first_8_bytes |> :binary.decode_unsigned(:little)
+  end
+
+  @spec uint_to_bytes4(integer()) :: SszTypes.bytes4()
+  defp uint_to_bytes4(value) do
+    # Converts an unsigned integer value to a bytes 4 value
+    <<value::32>> |> :binary.bin_to_list() |> Enum.reverse() |> :binary.list_to_bin()
```

Comment:
You can also encode binaries using pattern matching
```suggestion
    <<value::unsigned-integer-little-size(32)>>
```